README.md

```diff
@@ -1,2 +1,155 @@
+
+It consists of two main classes:
+
+1. `DataHubService` - stores and manages `DataHub` configs with a remote 
+  [Data Hub storage server](https://github.com/digitalbazaar/bedrock-data-hub-storage).
+  Used to create or fetch `DataHub` instances.
+2. `DataHub` - provides a CRUD (+ find) interface to a specific Data Hub. Also
+  ensures appropriate db indexes are set up, and provides some key management
+  under the hood.
+
+## Install
+
+To install locally (for development):
+
+```
+git clone https://github.com/digitalbazaar/bedrock-web-data-hub.git
+cd bedrock-web-data-hub
+npm install
+```
+
+(Optional) Also install test dependencies:
+
+```
+cd test
+npm install
+cd ..
+```
+
+## Usage
+
+### Creating and registering a DataHub config
+
+Setting up a Data Hub Service requires configuring a Key Management Service
+([`bedrock-web-kms`](https://github.com/digitalbazaar/bedrock-web-kms)) instance.
+
+```js
+import {AccountMasterKey, KmsService} from 'bedrock-web-kms';
+import {DataHub, DataHubService} from 'bedrock-web-data-hub';
+
+// First, create a Master Key (via a key management service)
+const kmsService = new KmsService();
+// TODO: Explain kmsPlugin and accountId
+const masterKey = await AccountMasterKey.fromSecret({secret, accountId, kmsService, kmsPlugin});
+
+// Use the Master Key to create KEK and HMAC keys
+const kek = await masterKey.generateKey({type: 'kek'}); // Key Encryption Key
+const hmac = await masterKey.generateKey({type: 'hmac'});
+
+// Create a `DataHubService` instance (which can be used to create DataHub instances)
+const dhs = new DataHubService();
+```
+
+Now you can create and register a new `DataHub` configuration:
+
+```js
+const controller = 'account id goes here';
+
+const primary = true; // TODO: Explain what a primary data hub is
+
+const config = {
+  sequence: 0,  // TODO: is sequence required?
+  controller,
+  primary,
+  kek: {id: kek.id, algorithm: kek.algorithm},
+  hmac: {id: hmac.id, algorithm: hmac.algorithm}
+};
+
+const remoteConfig = await dhs.create({config}); // sends a POST request to the remote service
+const hub = new DataHub({config: remoteConfig, kek, hmac});
+```
+
+### Loading a saved DataHub config
+
+If you have previously registered a DataHub config (via `create()`), and you
+know its `id`, you can fetch its config via `get()`:
+
+```js
+// previously registered config
+const {id} = await dhs.create({config});
+
+// later, it can be fetched via the id
+const remoteConfig = await dhs.get({id});
+const hub = new DataHub({config: remoteConfig, kek, hmac});
+```
+
+If you know a controller/`accountId` but do not know a specific hub `id`, you can
+request "primary registered data hub for a given account":
+
+```js
+const remoteConfig = await dhs.getPrimary({controller: accountId});
+const hub = new DataHub({config: remoteConfig, kek, hmac});
+```
+
+### Using a DataHub instance for document storage
+
+See the API section below.
+
+## API
```

Comment:
I imagine we'll eventually just link to a jsdoc generated file for this?

Reply:
Yeah, you're probably right.